### PR TITLE
Sprk 80 - [M] Add pcs validity and running check to wrappers

### PR DESCRIPTION
**Pull request by jeroenrook**
Addressed the following checks when adding a solver
- Check if the scripts that are required are present:  smac_sparkle_wrapper.py
- In case of the presence of a pcs file the pcs file needs to be parsed to see if it is correct.

After discussion we decided not to implement the following point, as there are too many scenarios that probing will not work on all of those. 
- Run a probing run on the wrappers to see if they output expected lines. We need to think if this is possible without having an instance. One possibility might be to pass along an unknown instance and check if the wrapper reports a crash.

### CODE REVIEW

**Comment by thijssnelleman** [comment]
I hope you checked that sparkle_solver_wrapper.py is present and not the smac one?

File: CLI/add_solver.py [hunk 1]
```diff
@@ -100,10 +115,12 @@ def parser_function() -> argparse.ArgumentParser:
         sys.exit(-1)
     shutil.copytree(solver_source, solver_directory, dirs_exist_ok=True)
 
+    # JG: This is not correct as there does not necessarily need to be a binary.
+    # JG: It is important that the wrapper python file has executable rights!
     # check if the solver binary in the given directory has execution permission
-    for f in Path(solver_directory).iterdir():
-        if f.is_file() and f.suffix == "":
-            sfh.check_file_is_executable(f)
+    # for f in Path(solver_directory).iterdir():
```
Comment (thijssnelleman) [inline CLI/add_solver.py:121]: Remove?

File: sparkle/solver/solver.py [hunk 2]
```diff
@@ -59,11 +60,40 @@ def get_pcs_file(self: Solver) -> Path:
...
-        return self.directory / file_name
+        return file_path
+
+    def read_pcs_file(self: Solver) -> None:
+        """Read the pcs file."""
+        pcs_file = self._get_pcs_file()
+        try:
+            parser = pcsparser.PCSParser()
+            parser.load(str(pcs_file), convention="smac")
+            print("Could read pcs file")
```
Comment (thijssnelleman) [inline sparkle/solver/solver.py:94]: Redundant?

**Review by thijssnelleman — CHANGES_REQUESTED** [review]
Looks good, small fixes needed

File: sparkle/solver/solver.py (hunk 2 above)
Comment (jeroenrook) [reply sparkle/solver/solver.py:94]: Not redundant, but can move printing to CLI/add_solver.py and change to function return to a bool

Comment (thijssnelleman) [reply sparkle/solver/solver.py]: Sounds good to me

File: CLI/add_solver.py [hunk 3]
```diff
@@ -115,13 +118,6 @@ def parser_function() -> argparse.ArgumentParser:
         sys.exit(-1)
     shutil.copytree(solver_source, solver_directory, dirs_exist_ok=True)
 
-    # JG: This is not correct as there does not necessarily need to be a binary.
-    # JG: It is important that the wrapper python file has executable rights!
-    # check if the solver binary in the given directory has execution permission
-    # for f in Path(solver_directory).iterdir():
```
Comment (thijssnelleman) [inline CLI/add_solver.py]: Remove

File: CLI/add_solver.py [hunk 4]
```diff
@@ -115,13 +118,6 @@ def parser_function() -> argparse.ArgumentParser:
...
-    #     if f.is_file() and f.suffix == "":
-    #         sfh.check_file_is_executable(f)
```
Comment (thijssnelleman) [inline CLI/add_solver.py]: Remove

**Review by thijssnelleman — CHANGES_REQUESTED** [review]
Remove dead code

File: CLI/add_solver.py (hunk 3 above)
Comment (jeroenrook) [reply CLI/add_solver.py]: It is removed already :)

**Pull request merged**